bundle/bundle.go

```diff
@@ -477,8 +504,14 @@ func (r *Reader) Read() (Bundle, error) {
 		}
 	}
 
+	if bundle.GetBundleType() == DeltaBundleType {
+		if len(bundle.Data) != 0 || len(bundle.Modules) != 0 || len(bundle.WasmModules) != 0 {
```

Comment:
nit: I'd probably break this into three seperate cases and then return specific errors for each.

Reply:
Done and added tests for each.